### PR TITLE
Day 11 Solution: Hex Ed

### PR DESCRIPTION
**Pull request by shnako**
## Summary
- Implemented hexagonal grid navigation using cube coordinates
- Part 1: Find shortest path distance from origin to child process final position (664 steps)
- Part 2: Track maximum distance reached during the entire journey (1447 steps)

## Technical Approach
- **Cube Coordinates**: Used cube coordinate system (x, y, z) where x + y + z = 0 for hexagonal grids
- **Distance Calculation**: Leveraged the property that distance in cube coordinates is max(|x|, |y|, |z|)
- **Direction Mapping**: Each hex direction (n, ne, se, s, sw, nw) maps to specific coordinate changes
- **Efficient Tracking**: Part 2 maintains running maximum distance during path traversal

## Test Coverage
- All example test cases from the puzzle description pass
- Verified against actual input producing correct answers (Part 1: 664, Part 2: 1447)
- Comprehensive test coverage for both parts including edge cases

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* **New Features**
  * Added support for Day 11: Hex Ed (Part 1 & Part 2) and expanded runnable day range to include Day 11.

* **Documentation**
  * Updated README progress table with Day 11 links.
  * Added Day 11 puzzle statement for reference.

* **Tests**
  * Added tests covering Day 11 examples and expected results (including full input assertions).

* **Chores**
  * Included Day 11 input data file.

### CODE REVIEW

**Comment by shnako** [comment]
@coderabbitai review

File: src/solutions/day11/mod.rs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+/// In cube coordinates, x + y + z = 0 always holds
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+struct HexPosition {
+    x: i32,
+    y: i32,
+    z: i32,
+}
+
+impl HexPosition {
+    fn new() -> Self {
```
[the remark this answers is not in the document]
Comment (shnako) [reply src/solutions/day11/mod.rs:14]: Thank you for the review! The formatting issues have been addressed in commit ab14446. I ran cargo fmt to fix all formatting inconsistencies, including the parse_input function that needed to be on a single line. The CI checks should now pass successfully.
